### PR TITLE
Rewrite docs regarding load balancing

### PR DESCRIPTION
**Pull request by chris-schra**
Make clear that the zones and regions have
to match the actual K8s labels.
Add convenience scripts.

[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by chris-makaio** [comment]
@googlebot I signed it!

File: content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md [hunk 1]
```diff
@@ -57,49 +59,23 @@ metadata:
...
   do \
     kubectl --context="$CTX" apply -f sample.yaml; \
+    for LOC in $(kubectl --context="$CTX" get nodes -o \
+      custom-columns=REGION:'{.metadata.labels.topology\.kubernetes\.io/region}',ZONE:'{.metadata.labels.topology\.kubernetes\.io/zone}' --no-headers  \
+      | uniq | sed -E 's/<none>/default/g' | sed -E 's/[ \t]+/\./g')
+    do \
+      ./@samples/helloworld/gen-helloworld.sh \
+        --version "$LOC" > "helloworld-${LOC}.yaml"; \
+        kubectl --context="$CTX" apply -f "helloworld-${LOC}.yaml"; \
+    done
```
Comment (rvennam) [inline content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:77]: This is terse and clever, but I'm afraid this make it less readable and more difficult to modify. Open to other opinions

Comment (chris-makaio) [reply content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:77]: I take that as a compliment :)
Of course you‘re right, but my point here is that I think it should be clear that we have to use the actual region & zone labels from the clusters in Istio to make load balancing work. I can split that commands, no problem...

related (different repo):
https://github.com/istio/istio/issues/30832

File: content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md [hunk 2]
```diff
@@ -36,10 +36,12 @@ The following environment variables will be used for the various contexts:
 Variable | Description
 -------- | -----------
 `CTX_PRIMARY` | The context used for applying configuration to the primary cluster.
-`CTX_R1_Z1` | The context used to interact with pods in `region1.zone1`.
-`CTX_R1_Z2` | The context used to interact with pods in `region1.zone2`.
-`CTX_R2_Z3` | The context used to interact with pods in `region2.zone3`.
-`CTX_R3_Z4` | The context used to interact with pods in `region3.zone4`.
+`CTX_WORKERS` | The contexts used to interact with pods in the worker clusters.
```
Comment (nmittler) [inline content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:39]: Using an array and relying on index positions seems generally less clear IMO. I think we should just revert this

File: content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md (hunk 1 above)
Comment (nmittler) [reply content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:77]: +1 ... we need to prioritize readability here.

**Comment by chris-makaio** [comment]
guys, I understand and accept that you don't like the PR 😆 

The point is that it's not obvious how all the values relate to each other. Again, we could just simply set the values outside an array, but the important thing is -  I think - that one can't just ANY names for regions and zones. If I read "region1" and "zone1" it looks to me like I just can use any value as long as it is the same when passed to gen-helloworld

**Comment by nmittler** [comment]
@chris-makaio The [intro page](https://istio.io/latest/docs/tasks/traffic-management/locality-load-balancing/) for the task does describe where these values from from (node labels). Does that help?

**Comment by craigbox** [comment]
Chris; could you modify the PR to just make clear in comments the things you found confusing?

**Review by Xunzhuo — COMMENTED** [review]
please fix lint erros below thanks!

./en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:62: MD009 Trailing spaces
./en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:63: MD009 Trailing spaces
./en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:51: MD031 Fenced code blocks should be surrounded by blank lines
./en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:60: MD031 Fenced code blocks should be surrounded by blank lines
./en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md:66: MD031 Fenced code blocks should be surrounded by blank lines

**Review by craigbox — COMMENTED** [review]
Thanks for taking a stab at this, @chris-schra. Given there wasn't any follow up from the comments the first time around, I'm going to close this; feel free to re-open if you wish.

**Pull request closed without merging**